### DISCUSSION TITLE
Is it possible to show LSB, MSB and name of a chosen Performance when using DAW?

### DISCUSSION THREAD

**Discussion by DutchBigEd** [probonopd/MiniDexed#672] [Q&A] [answered]
I'm working on a "box" which will contain 3 MiniDexed DX7's and a USB to MIDI controller (Teensy 4), for each MiniDexed an Oled screen.
The MiniDexed I've made so far can show the LSB, MSB and name of the performance, but I have to navigate by means of the rotary encoder.
Via my DAW (Ableton) I can change the Performance by setting SUB to 1 and use PGM to choose from the wanted Performances, and I can hear that the right Performance is playing.
My question is: Is it possible to show the LSB, MSB and Performance on the Oled display, just like with the rotary encoder?
Now when I start the MiniDexed it just shows MiniDexed TG1 -> thats all, and no matter what I send with my DAW, the display stays the same.
Am I doing something wrong or is this not implemented?

**Comment by Banana71** [comment]
Hi Ed, you're not doing anything wrong. That's a function I really want too. I would also like to have the option to change the bank by pressing and turning the encoder in the Performance Load menu.

**Reply by Banana71** [reply]
For all those lucky enough to have multiple miniDexeds, the following would be helpful:
https://github.com/probonopd/MiniDexed/pull/581#issuecomment-1924735513

**Reply by DutchBigEd** [reply]
![midi interface MIDI OX](https://github.com/probonopd/MiniDexed/assets/17274454/e794660c-7644-41d9-9e2c-6cbc0c6046ac)
Thanks, I use a Teensy 4 to control 3 MiniDexed via USB, this is possible because the Teensy 4 has multiple UART's and I assign an UART for each MiniDexed so I can use the same micro SD card(same Performances, same minidexed.ini, performance.ini, cmdline.txt and config.txt)  for each MiniDexed.

**Reply by Banana71** [reply]
> #581 (comment)

it's just an idea that hasn't been implemented yet

**Reply by DutchBigEd** [reply]
Any update on this question?

**Reply by soyersoyer** [reply]
Does it change when you open the performance > load menu?

**Reply by DutchBigEd** [reply]
When I use the DAW, and set the minidexed to Performance 》load and change the performance from the DAW, the sound changes but not the display. It stayes on
001:001      load
<default

**Reply by soyersoyer** [reply]
This is because the program change changes the Voice by default, you can see the change under tg1 > voice. If you want to change the Performance, set the PerformanceSelectChannel in [minidexed.ini](https://github.com/probonopd/MiniDexed/wiki/Files#midi).

**Reply by soyersoyer** [reply]
With [DreamDexed](https://github.com/DreamDexed/DreamDexed/), you can [set](https://github.com/DreamDexed/DreamDexed/blob/f79848579645b0a3012510c53ee57ce569c4565a/src/minidexed.ini#L20) the default screen to start with the Performance > Load menu item.

**Reply by DutchBigEd** [reply]
I will try that and report  back here.
But first I have to openup my case in order to change the .ini file.

**Reply by DutchBigEd** [reply]
I installed DreamDexed and my module starts up with:
001:001: [L]     Load
<Default
But still, if I change a Performance with my DAW, there is no change on the oled screen, but my sound does change according to the patch pdf.

**Reply by soyersoyer** [reply, marked as answer]
Try this also:

> This is because the program change changes the Voice by default, you can see the change under tg1 > voice. If you want to change the Performance, set the PerformanceSelectChannel in [minidexed.ini](https://github.com/probonopd/MiniDexed/wiki/Files#midi).

PerformanceSelectChannel=17

**Reply by DutchBigEd** [reply]
Thanks a LOT! it works, although I don't understand how...

**Reply by soyersoyer** [reply]
The voices and performances are different. Please read this [page](https://github.com/probonopd/MiniDexed/wiki#tg--tone-generator).

**Reply by DutchBigEd** [reply]
But.... now with the DreamDexed on the uSD card there is a nasty noise on my signal :(

**Reply by soyersoyer** [reply]
That's weird.  No noise with the latest MiniDexed?

**Reply by DutchBigEd** [reply]
No, no noise before I changed to DreamDexed :(

**Reply by Banana71** [reply]
Hi Ed, Hi Soyer, with DreamDexed I sometimes get cracklers (especially with performances that use all 8 TGs) when I change parameters with the editor.

**Reply by soyersoyer** [reply]
> No, no noise before I changed to DreamDexed :(

You can change it back at any time.
Which raspberry model are you using?

**Reply by DutchBigEd** [reply]
I have no time for now to inverstigate further.
Thank you for you time so far!

**Reply by soyersoyer** [reply]
If you look at the output with an oscilloscope, you can see this one:
https://github.com/probonopd/MiniDexed/commit/98a31df28ddac5086c10dadaa7015a49870f4e5f
I don't think it's audible, maybe adding a DC would be better?

**Reply by DutchBigEd** [reply]
<img width="800" height="480" alt="SDS00002" src="https://github.com/user-attachments/assets/c00f4939-b81f-4f91-8123-7c2a004ff45d" />

This is what get from my output .
It gets weirder.... some (not all) of the performances are more or less quiet if I select them with DAW or Rotary encoder, but after I hit a tone, the noise is back.

**Reply by soyersoyer** [reply]
Do I understand correctly that these are 32mV spikes every 2.64 ms? 2.64 ms is understandable (128/48000), but the 32mV spike for the 0-1 transition is surprising to me at 23 bits.
Are you using PCM5102?

Is it possible that after changing performance there is a DC component left and this workaround is not activated?
I want a scope too!:)

This workaround is to prevent the pcm5102 from going into full analog mute, because it was audible. This doesn't look good either. I think it can be replaced by adding a DC.

**Reply by DutchBigEd** [reply]
Yes, I use PCM5102A modules for my MiniDexed DX7's
I kept getting noise so I went back to MiniDexed_1277_2025-07-30-8f2c638 and this is the result:

<img width="800" height="480" alt="SDS00008" src="https://github.com/user-attachments/assets/02575139-8c3a-47ca-8a3b-3b43c12bd38b" />

This is at the headphone output of my mixer (as is the previous picture), the max. output is about 10 volt p/p The only thing is that I first have to set the Minidexed DX7's to Performance first but then I still see the Performance chosen by the DAW on the little Oled screen. But I can live with that :)

**Reply by soyersoyer** [reply]
Could you please check if there are any spikes on the other channel (left/right)? (Or add ChannelsSwapped=1 to the minidexed.ini)
Panning has been [fixed](https://github.com/DreamDexed/DreamDexed/commit/5efaf365eb95af274b3a997fa85c46eeb14b3c16) in the mixer so the channels are swapped compared to MiniDexed.

**Reply by DutchBigEd** [reply]
I'll be back Tomorrow...

**Reply by DutchBigEd** [reply]
<img width="800" height="480" alt="MiniDexed" src="https://github.com/user-attachments/assets/b0a25cd5-604a-476a-b6b3-95a8441cece4" />

<img width="800" height="480" alt="MiniDexed_Swap" src="https://github.com/user-attachments/assets/8f20bbca-fe35-4a9c-aade-1cd311af1d3a" />

<img width="800" height="480" alt="DreamDexed" src="https://github.com/user-attachments/assets/b7bdb7dc-40d2-42e7-8a99-a5a3a3b6bc8e" />

<img width="800" height="480" alt="DreamDexed_Swap" src="https://github.com/user-attachments/assets/05450ac2-27a2-4273-9db9-38415d8d96e7" />

From top to bottom: 
MiniDexed_1277_2025-07-30-8f2c638
MiniDexed_1277_2025-07-30-8f2c638 audio swapped
DreamDexed_87_2025-09-17-f798485
DreamDexed_87_2025-09-17-f798485 audio swapped
In these pictures, as in the previous ones: Yellow = left channel. purple = right channel.
It doesn't like much, the pulses in the DreamDexed version, but it is clearly audible. 

This is my minidexed.ini for the DreamDexed version:
[minidexed.zip](https://github.com/user-attachments/files/23119921/minidexed.zip)

**Reply by soyersoyer** [reply]
Thank you. I'll figure out the problem.

**Reply by soyersoyer** [reply]
I can't reproduce it. I measure 20 mV p/p at idle without spikes, and 300 mV p/p while playing with Max MasterVolume.
Your 10 V p/p seems like a lot.
Which performance is affected? (one is enough)
Could you try with another power supply?
Is this version also affected? https://github.com/DreamDexed/DreamDexed/actions/runs/16926498159/artifacts/3751169100

**Reply by DutchBigEd** [reply]
The 10v p/p is from my mixer, so I don't know what the output voltage of the dac is...
Connecting another power supply is a bit difficult since my minidexed modules are built-in in a 1/2 19" rack.
But since I went back to the Minidexed the systems are again without noise, I am Ok.
I will try the last suggestion with dreamdexed.

Edit:
I also have a Minidexed DX7 in a small box and I think I found the problem.
It's shameful to say but I think it has to do with the fact that my computer is causing the trouble: If I power the stand-alone version from my pc using the USB C cable there is noise and when I power it from a powerbank, using the same USB C cable, there is not.
I did used the version from your link.
Since I use a Teensy as midi controller to control 3 Minidexed DX7 modules in one 1/2 euro rack, there is always a computer to be used.
Still it's strange why there is noise when connecting to a PC in the DreamDexed version and not in the original Minidexed one...

**Reply by Kirtai** [reply]
I'm curious, when you say a 1/2 19" rack do you mean a 0.5U 19" case or a 10" (half rack) case?

**Reply by DutchBigEd** [reply]
![20251102_073645](https://github.com/user-attachments/assets/965c4425-abc2-4192-a70f-de00f669470c)

This is my setup ☝🏻😁

**Reply by Kirtai** [reply]
Ah, thanks. That looks pretty good :)

How does the Teensy controller work?

**Reply by DutchBigEd** [reply]
<img width="463" height="608" alt="midi interface MIDI OX" src="https://github.com/user-attachments/assets/0dcedb6b-fb70-4231-b8aa-ca10cea92c7d" />

I have slightly adapted the example program supplied by the Arduino IDE library for the Teensy. 
This Teensy used has 4 uarts and as you can see, it shows up as a native midi interface with 4 devices.
They can be used with a DAW, but It's also possible to connect a midi controller/keyboard to the front connector.
Also one of the uarts on the Teensy is used as DIN midi in/out (MIDIIN4 and MIDIOUT4)

**Reply by Kirtai** [reply]
Ah, so they're on different virtual cables.
Thanks :)
